pkg/apk/auth/chainguard.go

```diff
@@ -0,0 +1,49 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"chainguard.dev/sdk/sts"
+	"google.golang.org/api/idtoken"
+)
+
+// NewChainguardIdentityAuth returns an Authenticator that authorizes
+// requests as the given assumeable identity.
+//
+// The identity is a UIDP of a Chainguard Identity.
+// Issuer is usually https://issuer.enforce.dev.
+// Audience is usually https://apk.cgr.dev.
+func NewChainguardIdentityAuth(identity, issuer, audience string) Authenticator {
+	return authenticator{
+		id:  identity,
+		iss: issuer,
+		aud: audience,
+	}
+}
+
+type authenticator struct {
+	id, iss, aud string
+}
+
+func (a authenticator) AddAuth(ctx context.Context, req *http.Request) error {
+	if req.Host != strings.TrimPrefix(a.aud, "https://") {
```

Comment:
JFYI I think we expect `aud` to be `apk.cgr.dev` (sans scheme), but this actually allows it to be either, so this is technically fine as-is.